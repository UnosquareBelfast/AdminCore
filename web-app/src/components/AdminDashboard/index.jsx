import React from 'react';
import { Link } from 'react-router-dom';
import { InnerLayout } from './styled';

const AdminDashboard = () => {
  return (
    <InnerLayout>
      <h2>Admin Dashboard</h2>
      <div className="columnWrap">
        <div>
          <h3>Employees</h3>
          <Link to="/admin/employees">View Employees</Link>
        </div>
        <div>
          <h3>Holidays</h3>
          <Link to="/admin/holidays">View All Holidays</Link>
          <Link to="/admin/pendingHolidays">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Clients</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Contracts</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
      </div>
    </InnerLayout>
  );
};

export default AdminDashboard;
