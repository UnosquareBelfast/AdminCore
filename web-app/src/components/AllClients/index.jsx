import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { Table, Button } from '../common';
import { ActiveDot, CTA } from './styled';
import employeeStatus from '../../utilities/employeeStatus';

export const AllClients = ({
  clients,
  viewClient,
  archive,
  createNewClient,
}) => {
  let clientList;
  if (clients.length > 0) {
    clientList = clients.map(client => (
      <tr key={client.clientId}>
        <td>{client.clientName}</td>
        <td>{client.clientStatusDescription}</td>
        <td>
          <ActiveDot active={client.clientStatusId === employeeStatus.ACTIVE} />
          {client.clientStatusId === employeeStatus.ACTIVE
            ? 'Active'
            : 'Inactive'}
        </td>
        <td>
          <button onClick={() => viewClient(client.clientId)}>
            Edit Client
          </button>
          {client.clientStatusId === employeeStatus.ACTIVE ? (
            <button className="success" onClick={() => archive(client)}>
              Archive
            </button>
          ) : (
            <button className="error" onClick={() => archive(client, true)}>
              Activate
            </button>
          )}
        </td>
      </tr>
    ));
  } else {
    clientList = (
      <tr>
        <td>No Clients</td>
        <td />
        <td />
        <td />
      </tr>
    );
  }
  return (
    <Fragment>
      <h2>All Clients</h2>
      <Table
        tableHeaders={['Name', 'Description', 'Status', ' ']}
        tableRows={clientList}
      />
      <CTA>
        <Button onClick={() => createNewClient()} label="New Client" />
      </CTA>
    </Fragment>
  );
};

AllClients.propTypes = {
  createNewClient: PT.func,
  clients: PT.array,
  archive: PT.func.isRequired,
  viewClient: PT.func.isRequired,
};

export default container(AllClients);
