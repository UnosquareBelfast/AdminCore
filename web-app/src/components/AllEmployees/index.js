import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import { UserList, UserModal } from '../';
import { Button } from '../common';
import { CornerButton } from '../common_styled';
import container from './container';

const AllEmployees = ({ history, users, selectUser, selectedUser }) => {
  return (
    <Fragment>
      <CornerButton>
        <Button
          onClick={() => history.replace('/admin/employees/new')}
          label="New Employee"
        />
      </CornerButton>
      <UserModal
        user={selectedUser}
        closeModal={() => selectUser({})}
        history={history}
      />
      <h2>Employees</h2>
      <UserList
        history={history}
        users={users}
        columns={['fullName', 'email', 'role', 'location']}
        onRowClick={employee => selectUser(employee)}
      />
    </Fragment>
  );
};

AllEmployees.propTypes = {
  history: PT.object.isRequired,
  users: PT.array.isRequired,
  selectUser: PT.func.isRequired,
  selectedUser: PT.object.isRequired,
};

export default container(AllEmployees);
