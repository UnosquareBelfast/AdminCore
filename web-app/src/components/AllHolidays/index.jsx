import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { HolidayList, HolidayModal } from '../';

export const AllHolidays = ({ holidays, selectedHoliday, selectHoliday }) => {
  return (
    <Fragment>
      <HolidayModal
        holiday={selectedHoliday}
        closeModal={() => selectHoliday({})}
      />
      <h2>All Holidays</h2>
      <HolidayList
        holidays={holidays}
        columns={[
          'status',
          'employee',
          'startDate',
          'endDate',
          'requestedDate',
        ]}
        onRowClick={holiday => selectHoliday(holiday)}
      />
    </Fragment>
  );
};

AllHolidays.propTypes = {
  holidays: PT.array.isRequired,
  selectedHoliday: PT.object.isRequired,
  selectHoliday: PT.func.isRequired,
};

export default container(AllHolidays);
