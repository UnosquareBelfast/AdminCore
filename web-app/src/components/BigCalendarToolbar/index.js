import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { StyleContainer } from './styled';

const BigCalendarToolbar = props => {
  const goToBack = () => {
    props.onNavigate('PREV');
  };

  const goToNext = () => {
    props.onNavigate('NEXT');
  };

  const goToToday = () => {
    props.onNavigate('TODAY');
  };

  return (
    <StyleContainer>
      <div>
        <h2>Booking Dashboard</h2>
        <p className="date">{props.label}</p>
      </div>
      <div>
        <button onClick={goToBack}>Prev Month</button>
        <button onClick={goToToday}>This Month</button>
        <button onClick={goToNext}>Next Month</button>
      </div>
    </StyleContainer>
  );
};

BigCalendarToolbar.propTypes = {
  label: PT.string.isRequired,
  date: PT.object.isRequired,
  onNavigate: PT.func.isRequired,
};

export default BigCalendarToolbar;
