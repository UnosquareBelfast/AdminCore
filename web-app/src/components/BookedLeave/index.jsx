import React from 'react';
import { PropTypes as PT } from 'prop-types';
import styles from './style.css';
import container from './container';

export const BookedLeave = (props) => {
  const formatDate = (date) => {
    const definedDate = new Date(date);
    const year = definedDate.getFullYear();
    const month = definedDate.toLocaleString('en-us', { month: 'long'});
    const day = definedDate.getDate();

    return year + '-' + month + '-' + day;
  };

  return (
    <div>
      {
        props.bookedHolidays.map(holiday => {
          return <span className={styles.LeaveTrackerList}>{formatDate(holiday.date)}</span>;
        })
      }
    </div>
  );
};

BookedLeave.propTypes = {
  bookedHolidays: PT.array,
};

export default container(BookedLeave);
