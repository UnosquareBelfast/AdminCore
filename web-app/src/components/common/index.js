export { default as Card } from './Card';
export { default as Button } from './Button';
export { default as Modal } from './Modal';
export { default as RangeDayPicker } from './RangeDayPicker';
export { default as Errorbox } from './Errorbox';
export { default as Event } from './Event';
export { default as Input } from './Input';
export { default as Form } from './Form';
export { default as Steps } from './Steps';
export { default as Table } from './Table';
export { default as Filter } from './Filter';
export { default as Email } from './Email';
