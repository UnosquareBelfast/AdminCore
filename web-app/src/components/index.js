export { default as AdminDashboard } from './AdminDashboard';
export { default as AllClients } from './AllClients';
export { default as AllHolidays } from './AllHolidays';
export { default as BigCalendarToolbar } from './BigCalendarToolbar';
export { default as BookedLeave } from './BookedLeave';
export { default as BookingCalendar } from './BookingCalendar';
export { default as BookingModal } from './BookingModal';
export { default as CreateClient } from './CreateClient';
export { default as CreateContract } from './CreateContract';
export { default as ClientModal } from './ClientModal';
export { default as CreateTeam } from './CreateTeam';
export { default as CreateUser } from './CreateUser';
export { default as Legend } from './Legend';
export { default as PendingHolidays } from './PendingHolidays';
export { default as RangeDayPicker } from './common/RangeDayPicker';
export { default as RequestHoliday } from './RequestHoliday';
export { default as TakenLeave } from './TakenLeave';
export { default as TeamView } from './TeamView';
export { default as UserDetails } from './UserDetails';
export { default as UserModal } from './UserModal';
export { default as ViewContracts } from './ViewContracts';
export { default as ViewTeams } from './ViewTeams';
export { default as HolidayModal } from './HolidayModal';
export { default as AllEmployees } from './AllEmployees';
export { default as DataTable } from './DataTable';
