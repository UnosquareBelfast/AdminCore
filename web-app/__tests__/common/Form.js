import React from 'react';
import { shallow } from 'enzyme';
import { Form } from '../../src/components/common';

const mockHandleEvent = jest.fn();
const formData = {
  test: 'test',
};

xdescribe('Form Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Form
        formData={formData}
        actions={[
          {
            label: 'Test Form Name',
            event: mockHandleEvent,
            disabled: false,
          },
        ]}
      />,
    );
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  xit('on Form Submit handleSubmit is called', () => {
    const wrapper = shallow(<Form handleSubmit={mockHandleEvent} />);
    let form = wrapper.find('form');

    expect(mockHandleEvent.mock.calls.length).toEqual(0);
    form.simulate('submit');
    expect(mockHandleEvent.mock.calls.length).toEqual(1);
  });
});
