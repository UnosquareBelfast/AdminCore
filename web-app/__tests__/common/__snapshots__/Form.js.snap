// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders correctly 1`] = `
<form>
  <styled.div>
    <React.Fragment
      key="0"
    >
      <Button
        disabled={false}
        label="Test Form Name"
        onClick={[MockFunction]}
      />
    </React.Fragment>
  </styled.div>
</form>
`;
